test/mpi/Makefile.mtest

```diff
@@ -14,25 +14,20 @@
 ## The goal is to reduce the amount of boilerplate that needs to exist in each
 ## Makefile.am
 
-# AM_CPPFLAGS are used for C++ code as well
-AM_CPPFLAGS = -I$(top_builddir)/include -I$(top_srcdir)/include
-LDADD = $(top_builddir)/util/mtest.$(OBJEXT) $(top_builddir)/util/mtest_common.$(OBJEXT)
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/dtpools/include
+LDADD = libmtest.la $(top_builddir)/dtpools/src/.libs/libdtpools.la
```

Comment:
This doesn't look right.  I don't think you should be assuming anything in `.libs`.  I think the libtool intended way to do this is to add these dependency libraries to `executable_DEPENDENCIES`.

Reply:
Sorry, I should have said `executable_LDADD` to add the libraries + `executable_DEPENDENCIES` for dependency tracking.

Reply:
For example, in Hydra, we do this:

```
mpiexec_hydra_LDADD = -lpm -lhydra $(external_libs)
mpiexec_hydra_DEPENDENCIES = libpm.la libhydra.la
```